src/ripple/ledger/tests/Directory_test.cpp

```diff
@@ -0,0 +1,71 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/test/jtx.h>
+#include <ripple/ledger/Directory.h>
+
+namespace ripple {
+namespace test {
+
+struct Directory_test : public beast::unit_test::suite
+{
+    void testDirectory()
+    {
+        using namespace jtx;
+        Env env(*this);
+        auto gw = Account("gw");
+        auto USD = gw["USD"];
+        env.fund(XRP(10000), "alice", gw);
+
+        try
+        {
+            auto dir = Dir(*env.open(),
+                keylet::ownerDir(Account("alice")));
+            fail();
+        }
+        catch (...)
+        {
+            pass();
+        }
+
+        auto i = 10;
+        for (; i <= 400; i += 10)
+            env(offer("alice", USD(i), XRP(10)));
+
+        i = 0;
+        auto const dir = Dir(*env.open(),
+            keylet::ownerDir(Account("alice")));
```

Comment:
What's the point? A directory is a directory. Doesn't matter if its offers or owned objects.